staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go

```diff
@@ -578,6 +586,19 @@ func cleanVerb(verb, suggestedVerb string, request *http.Request) string {
 	return OtherRequestMethod
 }
 
+//getVerbIfWatch additionally ensures that GET or List would be transformed to WATCH
+func getVerbIfWatch(req *http.Request) string {
+	if strings.ToUpper(req.Method) == "GET" || strings.ToUpper(req.Method) == "LIST" {
+		// see apimachinery/pkg/runtime/conversion.go Convert_Slice_string_To_bool
```

Comment:
Let's split this part into a separate function:

func checkIfWatch(req *http.Request) bool {
		if values := req.URL.Query()["watch"]; len(values) > 0 {
			if value := strings.ToLower(values[0]); value != "0" && value != "false" {
				return true
			}
		}
		return false
}
```

And use that function both here and in line 557

Reply:
done!